diff --git a/cc/toolchains/args/layering_check/BUILD b/cc/toolchains/args/layering_check/BUILD
--- a/cc/toolchains/args/layering_check/BUILD
+++ b/cc/toolchains/args/layering_check/BUILD
@@ -48,7 +48,7 @@ cc_feature(
 
 cc_args(
     name = "layering_check_args",
-    actions = ["@rules_cc//cc/toolchains/actions:compile_actions"],
+    actions = ["@rules_cc//cc/toolchains/actions:source_compile_actions"],
     args = [
         "-fmodules-strict-decluse",
         "-Wprivate-header",
